models/CourseModel.php

```diff
@@ -754,4 +754,71 @@ public static function get_review_progress_link( $course_id, $user_id ) {
 	public static function get_course_preview_image_placeholder() {
 		return tutor()->url . 'assets/images/placeholder.svg';
 	}
+
+	/**
+	 * Retrieve the courses or course bundles that a given coupon code applies to.
+	 *
+	 * This function fetches published courses or course bundles from the database
+	 * based on the specified type. For each course, it retrieves the course prices
+	 * and the course thumbnail URL. If the user has Tutor Pro, it additionally
+	 * retrieves the total number of courses in a course bundle.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @param string $applies_to The type of items the coupon applies to. Accepts 'specific_courses'
+	 *                           for individual courses or any other value for course bundles.
+	 *
+	 * @global wpdb $wpdb WordPress database abstraction object.
+	 *
+	 * @return array An array of course objects. Each course object contains:
+	 *               - int $id: The ID of the course.
+	 *               - string $title: The title of the course.
+	 *               - string $type: The post type of the course (e.g., 'courses', 'course-bundle').
+	 *               - float $price: The regular price of the course.
+	 *               - float $sale_price: The sale price of the course.
+	 *               - string $image: The URL of the course's thumbnail image.
+	 *               - int|null $total_courses: The total number of courses in the bundle
+	 *                                          (only if the user has Tutor Pro and the course type is 'course-bundle').
+	 */
+	public function get_coupon_applies_to_courses( string $applies_to ) {
+		global $wpdb;
+
+		$post_type = 'specific_courses' === $applies_to ? 'courses' : 'course-bundle';
+
+		$where = array(
+			'post_status' => 'publish',
+			'post_type'   => $post_type,
+		);
+
+		$courses = QueryHelper::get_all( $wpdb->posts, $where, 'ID' );
+
+		if ( tutor()->has_pro ) {
+			$bundle_model = new \TutorPro\CourseBundle\Models\BundleModel();
+		}
+
+		$final_data = array();
+
+		if ( ! empty( count( $courses ) ) ) {
```

Comment:
No need to check the `count ( $courses )`, `empty ( $courses ) ` can serve the purpose.

Reply:
done